code/modules/pai/pai.dm

```diff
@@ -0,0 +1,450 @@
+		"bat" = FALSE,
+		"butterfly" = FALSE,
+		"cat" = TRUE,
+		"corgi" = FALSE,
+		"crow" = TRUE,
+		"duffel" = TRUE,
+		"fox" = FALSE,
+		"hawk" = FALSE,
+		"lizard" = FALSE,
+		"monkey" = TRUE,
+		"mouse" = TRUE,
+		"rabbit" = TRUE,
+		"repairbot" = TRUE,
+	)
+	/// List of all available card overlays.
+	var/static/list/possible_overlays = list(
+		"null",
+		"angry",
+		"cat",
+		"extremely-happy",
+		"face",
+		"happy",
+		"laugh",
+		"off",
+		"sad",
+		"sunglasses",
+		"what"
+	)
+
+/mob/living/silicon/pai/add_sensors() //pAIs have to buy their HUDs
+	return
+
+/mob/living/silicon/pai/can_interact_with(atom/target)
+	if(target == signaler) // Bypass for signaler
+		return TRUE
+	if(target == modularInterface)
+		return TRUE
+	return ..()
+
+// See software.dm for Topic()
+/mob/living/silicon/pai/canUseTopic(atom/movable/movable, be_close = FALSE, no_dexterity = FALSE, no_tk = FALSE, need_hands = FALSE, floor_okay = FALSE)
+	// Resting is just an aesthetic feature for them.
+	return ..(movable, be_close, no_dexterity, no_tk, need_hands, TRUE)
+
+/mob/living/silicon/pai/Destroy()
+	QDEL_NULL(atmos_analyzer)
+	QDEL_NULL(camera)
+	QDEL_NULL(hacking_cable)
+	QDEL_NULL(host_scan)
+	QDEL_NULL(instrument)
+	QDEL_NULL(internal_gps)
+	QDEL_NULL(newscaster)
+	QDEL_NULL(signaler)
+	if(!QDELETED(card) && loc != card)
+		card.forceMove(drop_location())
+		// these are otherwise handled by paicard/handle_atom_del()
+		card.pai = null
+		card.emotion_icon = initial(card.emotion_icon)
+		card.update_appearance()
+	GLOB.pai_list.Remove(src)
+	return ..()
+
+/mob/living/silicon/pai/emag_act(mob/user)
+	handle_emag(user)
+
+/mob/living/silicon/pai/examine(mob/user)
+	. = ..()
+	. += "Its master ID string seems to be [master_name || "empty"]."
+
+/mob/living/silicon/pai/get_status_tab_items()
+	. += ..()
+	if(!stat)
+		. += text("Emitter Integrity: [emitter_health * (100 / emitter_max_health)].")
+	else
+		. += text("Systems nonfunctional.")
+
+/mob/living/silicon/pai/handle_atom_del(atom/deleting_atom)
+	if(deleting_atom == hacking_cable)
+		hacking_cable = null
+		if(!QDELETED(card))
+			card.update_appearance()
+	if(deleting_atom == atmos_analyzer)
+		atmos_analyzer = null
+	if(deleting_atom == camera)
+		camera = null
+	if(deleting_atom == host_scan)
+		host_scan = null
+	if(deleting_atom == internal_gps)
+		internal_gps = null
+	if(deleting_atom == instrument)
+		instrument = null
+	if(deleting_atom == newscaster)
+		newscaster = null
+	if(deleting_atom == signaler)
+		signaler = null
+	return ..()
+
+/mob/living/silicon/pai/Initialize(mapload)
+	var/obj/item/pai_card/pai_card = loc
+	START_PROCESSING(SSfastprocess, src)
+	GLOB.pai_list += src
+	make_laws()
+	for (var/law in laws.inherent)
+		lawcheck += law
+	if(!istype(pai_card)) //when manually spawning a pai, we create a card to put it into.
+		var/newcardloc = pai_card
+		pai_card = new(newcardloc)
+		pai_card.set_personality(src)
+	forceMove(pai_card)
+	card = pai_card
+	job = JOB_PERSONAL_AI
```

Comment:
Why does this all have to happen before calling the parent? If you're trying to run code before special code in the parent runs, you should consider adding a proc to the parent that is called before the special code is ran which you can then override to put all this data in. It leads to a lot less unintended side-effects as a result of executing code before the parent is called

Reply:
i actually don't know, this was here before me and I just kinda migrated it over without rewriting it. i guess now would be a good time to do so